Fix UX details in hamburger menu

andreslucena opened the pull request:
#### :tophat: What? Why?

This PR introduces a couple of mini changes based on #12932 

#### :pushpin: Related Issues
 
- Fixes [#13015](https://github.com/decidim/decidim/issues/13015)

#### Testing

Open the hamburguer menu in a mobile resolution. Check the Acceptance criteria from #13015:

- [x] Given the hamburger menu is opened, when the menu expands, then it should occupy 100% of the screen height.
- [ ] Given the hamburger menu is opened, when participants view the menu items, then the access to help pages (/pages) should be visible after the menu items.
- [x] Given the language selector is in the hamburger menu, when participants use the language selector, then it should have the same style as the locale configuration in the participant's account settings.

### :camera: Screenshots
 
![Changes introduced in this PR](https://github.com/decidim/decidim/assets/717367/76fed20e-cc50-4552-87e2-d2d813143d3f)

:hearts: Thank you!

andreslucena commented:
@carolromero I'm hesitant about this 

>  Given the language selector is in the hamburger menu, when participants use the language selector, then it should have the same style as the locale configuration in the participant's account settings.

As exactly the same style could be too big and too notorious. The thing is that the search bar does not have a border, and the locale switcher in the account settings is like this:

![Screenshot of the locale switcher in the account page](https://github.com/decidim/decidim/assets/717367/4f63afa2-a2a2-4f46-917a-055eda53acb4)

So, I removed the border so at least is a bit less noticeable. Let me know if you  want to tune it a bit (we can do a bit of pair programming to iterate faster)

alecslupu reviewed (changes requested):
@andreslucena , this is not Ok. 

The acceptance criteria states: " Given the hamburger menu is opened, when participants view the menu items, then the **access to help** pages (/pages) **should be visible after the menu items**.", or as it is implemented now, we loose all the menu items : 

Develop: 

![image](https://github.com/decidim/decidim/assets/105683/18f9d36e-007c-4bc2-b492-472fba99d17a)

This Branch: 

![image](https://github.com/decidim/decidim/assets/105683/8d97be26-44c9-42a6-a4e7-643c97ea07f6)

If we merge this, we loose mobile access to : 
- Processes
- Assemblies
- Initiatives 
- Consultations

andreslucena commented:
> If we merge this, we loose mobile access to (...)

Uooops, I was working in other PRs related with the seeds and I had a memory lapse after introducing the `:mobile_menu` change 🤦🏽 

It should be fixed in the last couple PRs: I introduced the spaces and also added a spec just in case. 

This is ready for another round.

andreslucena commented on decidim-core/app/packs/stylesheets/decidim/_header.scss:
```diff
@@ -24,10 +24,6 @@ header {
   }
 
-  #menu-bar-container {
-    @apply mt-24 md:mt-0;
```
This fixes a bug where there's an unnecessary margin-top. I only can reproduce it in nightly :/ 

![Screenshot of the margin-top in the header bug](https://github.com/decidim/decidim/assets/717367/87b59c12-e1ac-4f33-b639-1099db52e41a)

It doesn't seem to break anything so I think it's OK to merge this

alecslupu replied:
I am against this change, within this PR. 
The empty space that you see on nightly is caused by omnipresent banner. Removing the class above, without having omnipresent banner enabled, will cause other pages to missbehave: 

This is how the show proposal page looks like.
![image](https://github.com/decidim/decidim/assets/105683/c3e5c442-337c-4cc5-a6f1-e82b962afbf8)

alecslupu replied:
when omnipresent banner is displayed, and when the admin bar is being shown, the following content is rendered 
![image](https://github.com/decidim/decidim/assets/105683/7fbd431d-fe92-44b0-8516-b262c2ffb44c)

Adding some CSS styling to omnipresent-banner, we can see the following result:
```
  z-index: 50;
  position: fixed;
``` 
![image](https://github.com/decidim/decidim/assets/105683/5d8fb91d-c9be-4089-983f-0f702f6fdd8a)

andreslucena replied:
Oh that's why I couldn't reproduce it locally! I answered to this concern at https://github.com/decidim/decidim/pull/13025#issuecomment-2213091167

alecslupu reviewed (changes requested):
Let's add back the CSS for `#menu-bar-container`, and address the omnipresent-banner in a different PR.

andreslucena commented:
> when omnipresent banner is displayed, and when the admin bar is being shown, the following content is rendered

Nice catch! To be honest I didn't noticed this misbehavior of the omnipresent banner until you mentioned. It's one of those things were once you've seen it once, you can't not notice that its there 😄 

I went to another alternative for fixing this issue:

1. Moving this banner (and the others that we can have: offline and impersonations) to the sticky header
2. Adding a dynamic margin top to the container that's being covered by the sticky header, as we don't know if the margin top should be `mt-24` or other, as we could be an admin, have the omnipresent, and be offline at the same time (for instance).

alecslupu reviewed (approved):
LGTM
